core/arch/arm/kernel/kern.ld.S

```diff
@@ -50,13 +50,19 @@
 
 #include <platform_config.h>
 
+#ifndef SMALL_PAGE_SIZE
+#define SMALL_PAGE_SIZE		4096
+#endif
+
 OUTPUT_FORMAT(CFG_KERN_LINKER_FORMAT)
 OUTPUT_ARCH(CFG_KERN_LINKER_ARCH)
 
 ENTRY(_start)
 SECTIONS
 {
 	. = CFG_TEE_LOAD_ADDR;
+	. = ALIGN(SMALL_PAGE_SIZE);
+	__core_unpg_rx_start = .;
```

Comment:
you're right. `__core_unpg_rx_start` aims at being `__text_start`.
I would tend to keep definition of each `__core_(unpg|init)_r(x|o|w)_(start|end|size)` and add a `PROVIDE(__core_unpg_rx_start = __text_start)` but maybe you would prefer to not introduce the useless `_core_XXXX_rX_XXX` and use the already defined flag. I felt my way helped more understanding the mapping when reading 'kernel.ld.S'.